### DISCUSSION TITLE
Add option for setting default timezone and attribute indicating if default timezone is used

### DISCUSSION THREAD

**Discussion by morcef** [crsmithdev/arrow#1090] [Q&A]
Hi,
I just recently started looking into Arrow, and I like it.
However, I am having a little difficulty trying to get my head around solving my issue with timezones.
The problem I am facing is that I want to parse times from strings that might, or might not contain timezone information (user input and external APIs), and I would like them to end up in local timezone. This also means that any naïve times should be parsed as local (default to use local timezone).
To illustrate I will use the following examples:

```
>>> arrow.get("2022-01-01 12:00")
<Arrow [2022-01-01T12:00:00+00:00]>

>>> arrow.get("2022-01-01 12:00+00")
<Arrow [2022-01-01T12:00:00+00:00]>

>>> arrow.get("2022-01-01 12:00", tzinfo="local")
<Arrow [2022-01-01T12:00:00+01:00]>

>>> arrow.get("2022-01-01 12:00+00", tzinfo="local")
<Arrow [2022-01-01T12:00:00+01:00]>

>>> arrow.get("2022-01-01 12:00+00").to("local")
<Arrow [2022-01-01T13:00:00+01:00]>
```

In the first two examples I am not able to differentiate the inputs because UTC is used as a default. As I want the output to be in local timezone I tried to set `tzinfo="local"`, which works as expected for any input that does not contain timezone information, as seen in the third example. If the input contains timezone information, however, this actually replaces the tzinfo instead of converting the time, seen in example four. So, to get the correct time with an input containing timezone information I must use the `to()` method as seen in the last exmaple, and for naive inputs I have to use the `tzinfo` parameter.

Currently I cannot see that an Arrow object contains information about the input being naïve (if the default UTC timezone has been used), or the possibility to chose a default timezone to use for naïve inputs. I believe that both of these options would be nice to have, but I am open to any other suggestions to solving my problem.

For the first idea, it might be possible to add an attribute to the Arrow object that indicates if the default timezone has been used, which could be possible to set within this if statement: https://github.com/arrow-py/arrow/blob/f8a65212df1ab234013a8a116436b3532c82b05d/arrow/arrow.py#L323-L325
The second idea I guess would mean adding a new parameter to the `get()` method, such as `default_tz` or similar, that is passed on for usage in the same spot.

I will try to illustrate the ideas I am proposing:
```
>>> naive_input = arrow.get("2022-01-01 12:00", default_tz="local")
>>> naive_input
<Arrow [2022-01-01T12:00:00+01:00]>
>>> naive_input.used_default_tz
True

>>> aware_input = arrow.get("2022-01-01 12:00+00", default_tz="local")
>>> aware_input
<Arrow [2022-01-01T12:00:00+00:00]>
>>> aware_input.used_default_tz
False

>>> arrow.get("2022-01-01 12:00", default_tz="local").to("local")
<Arrow [2022-01-01T12:00:00+01:00]>

>>> arrow.get("2022-01-01 12:00+00", default_tz="local").to("local")
<Arrow [2022-01-01T13:00:00+01:00]>

>>> arrow.get("2022-01-01 12:00+01", default_tz="local").to("local")
<Arrow [2022-01-01T12:00:00+01:00]>
```

As seen in the first two illustrations, any naïve input will use the default_tz, while any existing timezone information from input is used.
With this functionality I am able to use the same logic for all my inputs, as seen in the last three illustrations.

Thank you very much!

**Comment by morcef** [comment]
I also noted [this comment](https://github.com/arrow-py/arrow/issues/479#issuecomment-414438639), which I totally agree with:
> Note: I think any implementation that allows the caller to specify a default timezone should still prioritize an explicit timezone in the string being parsed.

**Comment by fluffy-critter** [comment]
This is also something I'd like to see, as the current state has been very inconvenient for one of my projects; see https://github.com/PlaidWeb/Publ/issues/262
